src/_Derivatives/subexpressions.jl

```diff
@@ -88,4 +88,35 @@ function order_subexpressions(
     return order_filtered, individual_order
 end
 
-export list_subexpressions, order_subexpressions
+# An implementation of depth-first-search for topologically sorting the
```

Comment:
Why is this much simpler than `order_subexpressions`? My sense is that either it's missing some necessary complexity or `order_subexpressions` can be similarly simplified.

Reply:
Yes, I'm working on a PR that simplifies `order_subexpressions`.